build.py

```diff
@@ -1,9 +1,22 @@
 import subprocess
+import sysconfig
+
 from hatchling.builders.hooks.plugin.interface import BuildHookInterface
 
 
 class CustomBuildHook(BuildHookInterface):
     def initialize(self, version, build_data):
+        # Set the custom wheel tag
+        build_data["tag"] = self._get_wheel_tag()
+        print(f"Using custom wheel tag: {build_data['tag']}")
         # Run setup.py build_ext before main build
         subprocess.check_call(["python", "setup.py", "build_ext", "--inplace"])
         return super().initialize(version, build_data)
+
+    def _get_wheel_tag(self):
+        # Without the tag, the wheel will be named jcvi-0.0.0-py3-none-any.whl
+        platform_tag = sysconfig.get_platform().replace("-", "_").replace(".", "_")
+        python_version = sysconfig.get_python_version().replace(".", "")  # e.g., "310"
+        python_impl = "cp"  # Assuming CPython. Modify if using PyPy or others.
+        abi_tag = f"{python_impl}{python_version}"
```

Comment:
I think the duplicated implementation+version tag "cp312" is incorrect here. 

Second instance should be [Application Binary Interface (ABI) tag](https://packaging.python.org/en/latest/specifications/platform-compatibility-tags).

See [PEP427 ](https://peps.python.org/pep-0427/#file-name-convention) and [PEP425](https://peps.python.org/pep-0425).

Maybe abi3? I don't know how to automate ABI tag selection.

Example generated wheel names:
jcvi-1.4.25.dev6+gbfef2f34-cp312-cp312-linux_x86_64.whl
jcvi-1.4.25.dev6+gbfef2f34-cp312-cp312-macosx_10_9_x86_64.whl

Reply:
thanks. Calling a method from `wheel` package instead (but this adds a dependency at `build_system.requires`).